aspnetcore/native-aot/index.md

```diff
@@ -0,0 +1,183 @@
+Restoring C:\Code\Demos\MyFirstAotWebApi\MyFirstAotWebApi.csproj:
+  Determining projects to restore...
+  Restored C:\Code\Demos\MyFirstAotWebApi\MyFirstAotWebApi.csproj (in 302 ms).
+Restore succeeded.
+```
+
+Use the following command to verify an app can but published using .NET native AOT:
+
+```
+PS> dotnet publish
+MSBuild version 17.<version> for .NET
+  Determining projects to restore...
+  Restored C:\Code\Demos\MyFirstAotWebApi\MyFirstAotWebApi.csproj (in 241 ms).
+C:\Code\dotnet\aspnetcore\.dotnet\sdk\8.0.<version>\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIde
+ntifierInference.targets(287,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotne
+t-support-policy [C:\Code\Demos\MyFirstAotWebApi\MyFirstAotWebApi.csproj]
+  MyFirstAotWebApi -> C:\Code\Demos\MyFirstAotWebApi\bin\Release\net8.0\win-x64\MyFirstAotWebApi.dll
+  Generating native code
+  MyFirstAotWebApi -> C:\Code\Demos\MyFirstAotWebApi\bin\Release\net8.0\win-x64\publish\
+```
+
+Note: The preceding output my differ from what you see depending on the version of  .NET 8 used.
+Review the contents of output directory:
+
+```
+$ dir bin\Release\net8.0\win-x64\publish
+
+    Directory: C:\Code\Demos\MyFirstAotWebApi\bin\Release\net8.0\win-x64\publish
+
+Mode                 LastWriteTime         Length Name
+----                 -------------         ------ ----
+-a---          30/03/2023  1:41 PM        9480704 MyFirstAotWebApi.exe
+-a---          30/03/2023  1:41 PM       43044864 MyFirstAotWebApi.pdb
+```
+
+The executable is self-contained and doesn't require a .NET runtime to run. When launched it should behave as expected:
+
+```
+$ .\bin\Release\net8.0\win-x64\publish\MyFirstAotWebApi.exe
+info: Microsoft.Hosting.Lifetime[14]
+      Now listening on: http://localhost:5000
+info: Microsoft.Hosting.Lifetime[0]
+      Application started. Press Ctrl+C to shut down.
+info: Microsoft.Hosting.Lifetime[0]
+      Hosting environment: Production
+info: Microsoft.Hosting.Lifetime[0]
+      Content root path: C:\Code\Demos\MyFirstAotWebApi
+```
+
+The `Program.cs` source file contains some changes for publishing to .NET native AOT. A significant difference is that `Microsoft.AspNetCore.Builder.WebApplication.CreateSlimBuilder` is used to create the web application builder.  The `CreateSlimBuilder` method initializes the <xref:Microsoft.AspNetCore.Builder.WebApplicationBuilder> with the minimal ASP.NET Core features necessary to run an application.
+<!-- Update the preceding with the following when the .NET 8 API is published :
+<xref:Microsoft.AspNetCore.Builder.WebApplication.CreateBuilderSlim%2A>
+-->
+
+```csharp
+var builder = WebApplication.CreateSlimBuilder(args);
+```
+
+Because this template uses JSON to serialize responses, we need to explicitly provide a `JsonSerializerContext` which specifies the custom types that we need to serialize so that the JSON source generator knows what to produce code for.
```

Comment:
```suggestion
This template uses JSON to serialize responses, to enable JSON serialization with native AOT, we need to explicitly provide a `JsonSerializerContext` which specifies the custom types that we need to serialize so that the JSON source generator knows what to produce code for.
```

We should link to the JSON source generator docs https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation

Reply:
Done.